contracts/network/BancorNetwork.sol

```diff
@@ -715,7 +729,10 @@ contract BancorNetwork is IBancorNetwork, Upgradeable, ReentrancyGuardUpgradeabl
         uint256 maxSourceAmount,
         uint256 deadline,
         address beneficiary
-    ) external payable whenNotPaused only(_bancorArbitrage) returns (uint256) {
+    ) external payable whenNotPaused returns (uint256) {
+        if (!_feeExemptionWhitelist.contains(msg.sender)) {
+            revert AccessDenied();
+        }
```

Comment:
Perhaps this can be refactored to a modifier?

Reply:
refactored